### PR TITLE
[New arch] Remove file

### PR DESCRIPTION
**Pull request by abelgardep**
## Related Issues
Implements https://github.com/owncloud/android/issues/2864

Library PR (if needed): https://github.com/owncloud/android-library/pull/394
_____

### CODE REVIEW

File: owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/RemoveFileUseCase.kt
Comment (JuancaG05) [inline owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/RemoveFileUseCase.kt:40]: 👍

**Review by JuancaG05 — COMMENTED** [review]
I think the conversion from Java to Kotlin is ok, and the rest also looks fine. As in the android-library PR, let's wait for another review and its explicit approval 💯

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt:93]: Maybe is better to use a `TODO` with a `New arch` tag so you can filter in the future what's pending? 🍻

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt:100]: `files.first` ? 🍻

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt:101]: Use { } even in one line? 🍻

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt:1219]: Use a `when` here? 🍻

**Review by davigonz — APPROVED** [review]

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt:93]: I introduced fixmes when we migrated the database, so I can filter them to check what's missing to move to the new arch. The idea is basically the same as TODOS, so I'll keep it this way since I already started using fixmes. Anyway, I will consider your proposal for future refactors 👍

**Pull request merged**
